rest/route/test.go

```diff
@@ -217,3 +217,56 @@ func (tgh *testGetHandler) addDataToResponse(resp gimlet.Responder, testResult i
 
 	return nil
 }
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// GET  /tasks/{task_id}/tests/count
+
+type testCountGetHandler struct {
+	taskID    string
+	execution int
+}
+
+func makeFetchTestCountForTask() gimlet.RouteHandler {
+	return &testCountGetHandler{}
+}
+
+func (h *testCountGetHandler) Factory() gimlet.RouteHandler {
+	return &testCountGetHandler{}
+}
+
+func (h *testCountGetHandler) Parse(ctx context.Context, r *http.Request) error {
+	projCtx := MustHaveProjectContext(ctx)
+	if projCtx.Task == nil {
+		return gimlet.ErrorResponse{
+			Message:    "task not found",
+			StatusCode: http.StatusNotFound,
+		}
+	}
+	h.taskID = projCtx.Task.Id
+
+	var err error
+	vals := r.URL.Query()
+	execution := vals.Get("execution")
+
+	if execution != "" {
+		h.execution, err = strconv.Atoi(execution)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				Message:    "invalid execution",
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+	}
+
+	return nil
+}
+
+func (h *testCountGetHandler) Run(ctx context.Context) gimlet.Responder {
+	count, err := data.TestCountByTaskID(ctx, h.taskID, h.execution)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "database error"))
```

Comment:
This could be a DB or Cedar issue:
```go
errors.Wrap(err, "counting tests for task '%s' and execution %d", h.taskID, h.execution)
```

Reply:
I just return the error since the `rest/data` code takes care of setting the error message.